x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/components/builtin_alert_types/threshold/expression.tsx

```diff
@@ -0,0 +1,1011 @@
+    termField,
+    thresholdComparator,
+    threshold,
+    timeWindowSize,
+    timeWindowUnit,
+  } = alert.params;
+
+  const firstFieldOption = {
+    text: i18n.translate(
+      'xpack.triggersActionsUI.sections.alertAdd.threshold.timeFieldOptionLabel',
+      {
+        defaultMessage: 'Select a field',
+      }
+    ),
+    value: '',
+  };
+
+  const [aggTypePopoverOpen, setAggTypePopoverOpen] = useState(false);
+  const [indexPopoverOpen, setIndexPopoverOpen] = useState(false);
+  const [indexPatterns, setIndexPatterns] = useState([]);
+  const [esFields, setEsFields] = useState<Record<string, any>>([]);
+  const [indexOptions, setIndexOptions] = useState<IOption[]>([]);
+  const [timeFieldOptions, setTimeFieldOptions] = useState([firstFieldOption]);
+  const [isIndiciesLoading, setIsIndiciesLoading] = useState<boolean>(false);
+  const [alertThresholdPopoverOpen, setAlertThresholdPopoverOpen] = useState(false);
+  const [alertDurationPopoverOpen, setAlertDurationPopoverOpen] = useState(false);
+  const [aggFieldPopoverOpen, setAggFieldPopoverOpen] = useState(false);
+  const [groupByPopoverOpen, setGroupByPopoverOpen] = useState(false);
+
+  const andThresholdText = i18n.translate(
+    'xpack.triggersActionsUI.sections.alertAdd.threshold.andLabel',
+    {
+      defaultMessage: 'AND',
+    }
+  );
+
+  const hasExpressionErrors = !!Object.keys(errors).find(
+    errorKey => expressionFieldsWithValidation.includes(errorKey) && errors[errorKey].length >= 1
+  );
+
+  const getIndexPatterns = async () => {
+    const indexPatternObjects = await loadIndexPatterns();
+    const titles = indexPatternObjects.map((indexPattern: any) => indexPattern.attributes.title);
+    setIndexPatterns(titles);
+  };
+
+  const expressionErrorMessage = i18n.translate(
+    'xpack.triggersActionsUI.sections.alertAdd.threshold.fixErrorInExpressionBelowValidationMessage',
+    {
+      defaultMessage: 'Expression contains errors.',
+    }
+  );
+
+  const setDefaultExpressionValues = () => {
+    setAlertProperty('params', {
+      aggType: DEFAULT_VALUES.AGGREGATION_TYPE,
+      termSize: DEFAULT_VALUES.TERM_SIZE,
+      thresholdComparator: DEFAULT_VALUES.THRESHOLD_COMPARATOR,
+      timeWindowSize: DEFAULT_VALUES.TIME_WINDOW_SIZE,
+      timeWindowUnit: DEFAULT_VALUES.TIME_WINDOW_UNIT,
+      triggerIntervalUnit: DEFAULT_VALUES.TRIGGER_INTERVAL_UNIT,
+      groupBy: DEFAULT_VALUES.GROUP_BY,
+      threshold: DEFAULT_VALUES.THRESHOLD,
+    });
+  };
+
+  const getFields = async (indexes: string[]) => {
+    return await getThresholdAlertTypeFields({ indexes, http });
+  };
+
+  useEffect(() => {
+    getIndexPatterns();
+  }, []);
+
+  useEffect(() => {
+    setDefaultExpressionValues();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  interface IOption {
+    label: string;
+    options: Array<{ value: string; label: string }>;
+  }
+
+  const getIndexOptions = async (pattern: string, indexPatternsParam: string[]) => {
```

Comment:
Is this tested in Functional tests of any kind?
Because, unless I'm missing something ,there are no unit tests around this.

Reply:
It's not tested yet at all, because development is in progress - this code should be skipped in the review and it won't be a visible part of the 7.6 UI